hxmath/math/DualQuaternion.hx

```diff
@@ -0,0 +1,622 @@
+     * 
+     * @param s         The scale value
+     * @return          The Scalar DualQuaternion
+     */
+    public static inline function scalar(s: Float):DualQuaternion
+    {
+        return new DualQuaternion(Quaternion.scalarReal( s ), Quaternion.zero);
+    }
+    
+    /**
+     * Clone.
+     * 
+     * @return          The cloned object.
+     */
+    public inline function clone():DualQuaternion
+    {
+        return new DualQuaternion( this.real.clone(), this.dual.clone() );
+    }
+    
+    /**
+     * dot product
+     *
+     * @param a         one DualQuaternion
+     * @param b         another DualQuaternion
+     * @return          the dot product.
+     */
+    public static inline function dot(a:DualQuaternion, b:DualQuaternion):Float
+    {
+        return Quaternion.dot(a.real.clone(), b.real.clone());
+    }
+    
+    /**
+     * Create a complex conjugate copy of a dualquaternion
+     * 
+     * @param a
+     * @return       ~a
+     */
+    @:op(~A)
+    public static inline function complexConjugate(a: DualQuaternion):DualQuaternion
+    {
+        return new DualQuaternion(~a.real, ~a.dual);
+    }
+    
+    /**
+     * Create a negated copy of a dualquaternion.
+     * 
+     * @param a
+     * @return      -a
+     */
+    @:op(-A)
+    public static inline function negate(a: DualQuaternion):DualQuaternion
+    {
+        return new DualQuaternion(-a.real, -a.dual);
+    }
+    
+    /**
+     * check if equal
+     * 
+     * @param a
+     * @param b
+     * @return       a==b
+     */
+    @:op( A == B )
+    public static inline
+    function equal(a:DualQuaternion, b:DualQuaternion):Bool
+    {
+       return (a == null && b == null) ||
+              a != null &&
+              b != null &&
+              a.real == b.real &&
+              a.dual == b.dual;
+    }
+    
+    /**
+     * check if not equal
+     * 
+     * @param a
+     * @param b
+     * @return       a!==b
+     */
+    @:op(A != B)
+    public static inline
+    function notEqual(a:DualQuaternion, b:DualQuaternion):Bool
+    {
+        return !equal( a, b );
+    }
+    
+    /**
+     * multiply by constant see also applyScale
+     * 
+     * @param a
+     * @param v
+     * @return       a*v
+     */
+    @:op(A * B)
+    @:commutative 
+    public static inline 
+    function scaleMultiply(a:DualQuaternion, v:Float):DualQuaternion
+    {
+        return new DualQuaternion(a.real * v, a.dual * v);
+    }
+    
+    /**
+     * divide by constant
+     * 
+     * @param a
+     * @param v
+     * @return       a/v
+     */
+    @:op(A / B)
+    public static inline
+    function divide(a:DualQuaternion, v:Float):DualQuaternion
+    {
+        return a * ( 1 / v );
+    }
+    
+    /**
+     * applyScale
+     * 
+     * @param a
+     * @return       scaled 
+     */
+    public inline function applyScale(s:Float):DualQuaternion
+    {
+        var cloned = clone();
+        cloned = cloned*DualQuaternion.scalar( s );
```

Comment:
Why not just call scaleMultiply? Unless scalar() is inlined, this will create an intermediate DualQuaternion.